templates/keepalived.conf.j2

```diff
@@ -50,6 +50,7 @@ vrrp_script {{ name }} {
   interval {{ details.interval | default(5) }}   # checking every {{ details.interval | default(5) }} seconds (default: 5 seconds)
   fall {{ details.fall | default(3) }}           # require {{ details.fall | default(3) }} failures for KO (default: 3)
   rise {{ details.rise | default(6) }}           # require {{ details.rise | default(6) }} successes for OK (default: 6)
+  weight {{ details.weight | default(0) }}       # adjust priority by this weight
```

Comment:
Good catch, we never defined it!

The code is good: the default matches the standard defaults.

However, I would prefer if this was wrapped with `{% if details.weight is defined %}`.

Without this conditional, running this role in an existing production environment would result in the addition of "weight 0" in the template, and restart keepalived. This could cause downtime if the role is not run in a serial way (yes, I know, I am VERY careful...).

Could you add conditionals and test if the template doesn't change if weight is not provided please?

Thank you in advance.

Reply:
Good point! I wasn't thinking about the restart of keepalived and possible outcomes.

Added if statement and tested it - no restart if no `weight` was defined 👍